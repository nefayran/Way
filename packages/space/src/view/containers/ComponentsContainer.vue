<template>
  <div class="wb-container">
    <div class="wb-components my-md mx-md pa-lg br-lg flex">
      <div class="wb-component">
        <component
          :is="wbComponent.component"
          v-if="wbComponent && wbComponent.component"
          v-model="value"
          v-bind="properties"
        />
        <h3 v-else class="font-secondary">EMPTY</h3>
      </div>
    </div>
  </div>
</template>

<script async setup lang="ts">
import { computed, ref, watchEffect, reactive } from "vue";
import store from "@/app/store";

const wbComponent = computed(
  () => store.getters["ComponentsStore/getComponent"]
);

const value = ref("@way");

// const properties = ref({
//   color: "primary--base",
// });

const properties = ref({});

watchEffect(async () => {
  if (wbComponent.value && wbComponent.value.props) {
    const props = await wbComponent.value.props;
    const propertiesObject = <Object>{};
    Object.keys(props).forEach((key: string) => {
      // Object.defineProperty(propertiesObject, key, {
      //   value: props[key],
      // });
      Object.defineProperty(propertiesObject, key, {
        value: props[key].default,
      });
      // propertiesObject[key] = props[key].default;
    });
    properties.value = propertiesObject;
  }
});
</script>

<style lang="scss">
.wb-components {
  height: calc(100% - var(--space-lg) * 3);
}

.wb-component {
  margin: auto;
}
</style>
