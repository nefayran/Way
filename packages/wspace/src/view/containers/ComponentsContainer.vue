<template>
  <div class="wb-container">
    <div class="wb-components my-md mx-md pa-lg br-lg flex">
      <div class="wb-component">
        <component
          :is="wbComponent.component"
          v-if="wbComponent && wbComponent.component"
        />
        <h3 v-else class="font-secondary">EMPTY</h3>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed } from "vue";
import store from "@/app/store";

const wbComponent = computed(
  () => store.getters["ComponentsStore/getComponent"]
);
</script>

<style lang="scss">
.wb-components {
  height: calc(100% - var(--space-lg) * 3);
  //background: linear-gradient(225deg, #000e3d, #001149);
  //box-shadow: 0 0 25px #000a29, 0 0 50px #00165f;
}

.wb-component {
  margin: auto;
}
</style>
