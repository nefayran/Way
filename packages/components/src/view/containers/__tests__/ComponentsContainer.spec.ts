import { mount } from "@vue/test-utils";
import ComponentsContainer from "../ComponentsContainer.vue";

describe("ComponentsContainer.vue", () => {
  test("should have an wb-container class", () => {
    const wrapper = mount(ComponentsContainer, {});
    expect(wrapper.classes()).toContain("wb-container");
    expect(wrapper.element).toMatchSnapshot();
  });
});
