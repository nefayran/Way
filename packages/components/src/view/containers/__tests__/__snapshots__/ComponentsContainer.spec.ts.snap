// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentsContainer.vue should have an wb-container class 1`] = `
<div
  class="wb-container"
>
  <div
    class="wb-components my-md mx-md pa-lg br-lg flex"
  >
    <div
      class="wb-component"
    >
      <h3
        class="font-secondary"
      >
        EMPTY
      </h3>
    </div>
  </div>
</div>
`;
